src/status_im/ui/screens/wallet/send/events.cljs

```diff
@@ -51,6 +89,57 @@
    (utils/set-timeout #(utils/show-popup (i18n/label :t/transaction-failed) message) 1000)))
 
 ;;;; Handlers
+;; HANDLE QR CODE
+
+(defn qr-data->send-tx-data [{:keys [address value symbol gas gasPrice public-key from-chat?]}]
+  {:pre [(not (nil? address))]}
+  (cond-> {:to address :public-key public-key}
+    value (assoc :amount value)
+    symbol (assoc :symbol symbol)
+    gas (assoc :gas (money/bignumber gas))
+    from-chat? (assoc :from-chat? from-chat?)
+    gasPrice (assoc :gas-price (money/bignumber gasPrice))))
+
+(defn extract-qr-code-details [chain qr-uri]
+  {:pre [(keyword? chain) (string? qr-uri)]}
+  ;; i don't like fetching all tokens here
+  (let [{:keys [:wallet/all-tokens]} @re-frame.db/app-db
+        qr-uri (string/trim qr-uri)
+        chain-id (ethereum/chain-keyword->chain-id chain)]
+    (or (let [m (eip681/parse-uri qr-uri)]
+          (merge m (eip681/extract-request-details m all-tokens)))
+        (when (ethereum/address? qr-uri)
+          {:address qr-uri :chain-id chain-id}))))
+
+(defn qr-data->transaction-data [qr-data]
+  {:pre [(map? qr-data)]}
+  ;; i don't like fetching all tokens here
+  (let [{:keys [:contacts/contacts]} @re-frame.db/app-db
```

Comment:
Same question as @dmitryn

Reply:
because there was strictly speaking no reason to have it through a sub, since the view didn't need to react on it's change